### PR TITLE
First attempt fixing null properties of Values<>

### PR DESCRIPTION
**Pull request by nickevansuk**
@RehanSaeed I've had a go at refactoring along the lines you explained, but all nulls now seem to be rendering... for both `OneOrMany<>` and `Values<>`!

Did I miss something here?

### CODE REVIEW

File: Source/Schema.NET/Values{T1,T2,T3,T4}.cs [hunk 1]
```diff
@@ -142,6 +142,11 @@ public Values(IEnumerable<object> items)
         /// </summary>
         public bool HasValue4 { get; }
 
+        /// <summary>
+        /// Gets whether the value of either type <typeparamref name="T1" />, type <typeparamref name="T2" />, type <typeparamref name="T3" />, or type <typeparamref name="T4" /> has a value.
+        /// </summary>
+        public bool HasValue => this.HasValue1 || this.HasValue2 || this.HasValue3 || this.HasValue4;
```
Comment (RehanSaeed) [inline Source/Schema.NET/Values{T1,T2,T3,T4}.cs]: Nice idea. This should be another PR. Looks like we might not get this one in quickly.

**Comment by RehanSaeed** [comment]
Yes, I tried the same and got the same weird result. Needs more investigation.

File: Source/Schema.NET/Values{T1,T2,T3,T4}.cs (hunk 1 above)
Comment (nickevansuk) [reply Source/Schema.NET/Values{T1,T2,T3,T4}.cs]: Ye this was mainly so the tests would continue passing, as quite a few relied on Nullable<>.HasValue, and broke when we removed the `?`

**Comment by nickevansuk** [comment]
Just to note that I'm on leave for the next few days, but any ideas from anyone on this very welcome!

**Comment by RehanSaeed** [comment]
Added your `HasValue` idea to master.

**Comment by RehanSaeed** [comment]
Closing in favour of https://github.com/RehanSaeed/Schema.NET/pull/84. Thanks for your efforts!
